notify/msteamsv2/msteamsv2.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2024 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package msteamsv2
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	commoncfg "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+
+	"github.com/prometheus/alertmanager/config"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+)
+
+const (
+	colorRed   = "Attention"
+	colorGreen = "Good"
+	colorGrey  = "Warning"
+)
+
+type Notifier struct {
+	conf         *config.MSTeamsV2Config
+	tmpl         *template.Template
+	logger       log.Logger
+	client       *http.Client
+	retrier      *notify.Retrier
+	webhookURL   *config.SecretURL
+	postJSONFunc func(ctx context.Context, client *http.Client, url string, body io.Reader) (*http.Response, error)
+}
+
+// https://learn.microsoft.com/en-us/connectors/teams/?tabs=text1#adaptivecarditemschema
+type Content struct {
+	Schema  string `json:"$schema"`
+	Type    string `json:"type"`
+	Version string `json:"version"`
+	Body    []Body `json:"body"`
+}
+
+type Body struct {
+	Type   string `json:"type"`
+	Text   string `json:"text"`
+	Weight string `json:"weigth,omitempty"`
+	Size   string `json:"size,omitempty"`
+	Wrap   bool   `json:"wrap,omitempty"`
+	Style  string `json:"style,omitempty"`
+	Color  string `json:"color,omitempty"`
+}
+
+type Attachment struct {
+	ContentType string  `json:"contentType"`
+	ContentURL  *string `json:"contentUrl"` // Use a pointer to handle null values
```

Comment:
Can't this just be a `string`? We know if the value is missing because `ContentURL == ""`?

Reply:
In the [adaptive cards documentation](https://learn.microsoft.com/en-us/connectors/teams/?tabs=text1%2Cdotnet#request-body-schema) - it says this value should be null: 

```
contentUrl | contentUrl | yes | string | Value should always be null
-- | -- | -- | -- | --
```

I reckon we can leave it as is.